gef.py

```diff
@@ -689,10 +689,30 @@ def size(self) -> int:
             raise AttributeError
         return self.page_end - self.page_start
 
+    def search_for_realpath(self) -> str:
+        # This function is a workaround for gdb bug #23764
+        # path might be wrong for remote sessions, so try a simple search for files
+        # that aren't found at the path indicated, which should be canonical.
+
+        # First, try the canonical path in the remote session root.
+        candidate1 = f"{gef.session.remote.root}/{self.path}"
+        if pathlib.Path(candidate1).is_file():
+            return candidate1
+
+        # On some systems, /lib(64) might be a symlink to /usr/lib(64), so try removing
+        # the /usr prefix.
+        if self.path.startswith("/usr"):
+            candidate = f"{gef.session.remote.root}/{self.path[4:]}"
```

Comment:
better to use trim instead of hardcoding the number of characters.

Reply:
Same idea with building the path here.

Reply:
Trim?

I can certainly use `len('/usr')` if that's preferred.

Reply:
Updated this using pathlib, too.